interfaces/configfiles/spec.go

```diff
@@ -0,0 +1,102 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configfiles
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/snap"
+)
+
+// Specification assists in collecting paths and content associated with an
+// interface.
+//
+// Unlike the Backend itself (which is stateless and non-persistent) this type
+// holds internal state that is used by the configfiles backend during the
+// interface setup process.
+type Specification struct {
+	// pathContent is the a map from absolute file paths to their expected content
+	pathContent map[string]string
+}
+
+// Methods called by interfaces
+
+// AddPathContent adds a configuration file with its content to the specification.
+func (spec *Specification) AddPathContent(path, content string) error {
+	if spec.pathContent == nil {
```

Comment:
We should ensure the path is clean before doing anything with it.

Reply:
Not sure what you mean here, note that a new spec is created each time `Setup()` is called, so in principle there is no need to clean-up the map.

Reply:
he means that path == Abs(path)   we have such checks in many places related to interface code and paths